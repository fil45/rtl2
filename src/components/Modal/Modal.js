import React, { useEffect } from "react";
import ReactDOM from "react-dom";

export default function Modal({
  children,
  isOpen = false,
  className = "root-modal",
  el = "div",
}) {
  const [container] = React.useState(document.createElement(el));

  useEffect(() => {
    document.body.appendChild(container);
    container.classList.add(className);
    return () => {
      document.body.removeChild(container);
    };
  }, [container, className]);

  return isOpen && ReactDOM.createPortal(children, container);
}
