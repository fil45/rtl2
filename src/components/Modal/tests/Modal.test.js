import React from "react";
import Modal from "../Modal";
import { render } from "@testing-library/react";
const somePhrase = "test div for test";
const testedComponent = (
  <Modal isOpen={true} children={<div>{somePhrase}</div>} />
);

describe("Modal component --->", () => {
  it("should render the page", () => {
    const { getByText } = render(testedComponent);
    const childrenInsideModal = getByText(somePhrase);
    expect(childrenInsideModal).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(testedComponent);
    expect(asFragment(testedComponent)).toMatchSnapshot();
  });
});
