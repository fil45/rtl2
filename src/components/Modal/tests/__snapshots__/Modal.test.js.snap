// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component ---> should match snapshot 1`] = `<DocumentFragment />`;
